### PR TITLE
fix(curriculum): improve Library Manager tests

### PR DESCRIPTION
**Pull request by gikf**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

- - -
- Tests ensure camper is using array passed in to functions.
- Since tests rely mainly on the returned result, there's no need to test parameter names.
- Removes from Step 4 test that appears to be incomplete/obsolete.
- Slight wording changes related to first two points.

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c1.md [hunk 1]
```diff
@@ -11,7 +11,7 @@ The rest of the objects representing the books have been filled in for you. You
 
 Now, you should start working on getting several common parts of the books.
 
-Start by displaying all the books. Create a `displayBooks` function with a `catalog` parameter. The function should return a string that contains the `title`, `author`, and `pages` of all the books in the `library` array.
+Start by displaying all the books. Create a `displayBooks` function with parameter accepting array with book objects. The function should return a string that contains the `title`, `author`, and `pages` of all the books in array passed to function.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c1.md:14]:
```suggestion
Start by displaying all the books. Create a `displayBooks` function with a parameter accepting an array with book objects. The function should return a string that contains the `title`, `author`, and `pages` of all the books in the array passed to function.
```

File: curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c2.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ dashedName: step-5
 
 To display the book summaries, you can use the `about` property of each book.
 
-Create a `getBookSummaries` function with a `catalog` parameter. The `getBookSummaries` function should return an array containing all the `about` property value of each book.
+Create a `getBookSummaries` function with single parameter, accepting array with book objects. The `getBookSummaries` function should return an array containing all the `about` property value of each book.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c2.md:12]:
```suggestion
Create a `getBookSummaries` function with a single parameter, accepting an array with book objects. The `getBookSummaries` function should return an array containing all the `about` property value of each book.
```

File: curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c2.md [hunk 3]
```diff
@@ -37,17 +37,59 @@ Your `getBookSummaries` function should use a higher order function. Ex. (`filte
 assert.match(getBookSummaries.toString(), /filter|map|reduce|forEach/);
 ```
 
-Your `getBookSummaries` function should contain the total of `8` books available in the library.
+Your `getBookSummaries` function should return an array containing the same number of elements, as in array passed to it.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c2.md:40]:
```suggestion
Your `getBookSummaries` function should return an array with the same length as the array passed to it.
```

File: curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c2.md [hunk 4]
```diff
@@ -37,17 +37,59 @@ Your `getBookSummaries` function should use a higher order function. Ex. (`filte
...
+    author: 'Author C',
+    about: 'About C',
+    pages: 304,
+  },
+];
+assert.lengthOf(getBookSummaries(_differentLibrary), 3);
 ```
 
-Your `getBookSummaries` function should have the `about` property of each book in the library.
+Your `getBookSummaries` function should return array with `about` property of each book from the array passed to it.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c2.md:68]:
```suggestion
Your `getBookSummaries` function should return an array with the `about` property of each book from the array passed to it.
```

File: curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c3.md [hunk 5]
```diff
@@ -9,7 +9,7 @@ dashedName: step-6
 
 Another useful thing to do is to get the books by their author.
 
-Create a `getBooksByAuthor` function with a `catalog` and `author` parameters.
+Create a `getBooksByAuthor` function with two parameters - an array with book objects and string with author.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c3.md:12]:
```suggestion
Create a `getBooksByAuthor` function with two parameters - an array with book objects and a string with the author.
```

File: curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c4.md [hunk 6]
```diff
@@ -7,7 +7,7 @@ dashedName: step-7
 
 # --description--
 
-Finally, create a `getTotalPages` function with a `catalog` parameter. The function should return the total number of all the pages of the books in the `library` array.
+Finally, create a `getTotalPages` function with a single parameter, accepting array with book objects. The function should return the total number of pages in the books from array passed to function.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c4.md:10]:
```suggestion
Finally, create a `getTotalPages` function with a single parameter, accepting an array with book objects. The function should return the total number of pages in the books from the array passed to the function.
```

File: curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c4.md [hunk 7]
```diff
@@ -30,10 +30,32 @@ Your `getTotalPages` function should return a number.
 assert.isNumber(getTotalPages(library))
 ```
 
-Your `getTotalPages` function should return the number `2512`.
+Your `getTotalPages` function should return total number of pages in the books from array passed to it.
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/workshop-library-manager/67116d7584d0b469b14579c4.md:33]:
```suggestion
Your `getTotalPages` function should return the total number of pages in the books from the array passed to it.
```

**Review by majestic-owl448 — COMMENTED** [review]
The tests are good, I left some wording suggestions

**Review by majestic-owl448 — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
🚀

**Pull request merged**